bundler/lib/bundler/spec_set.rb

```diff
@@ -24,7 +24,7 @@ def for(dependencies, check = false, match_current_platform = false)
 
         specs_for_dep = spec_for_dependency(dep, match_current_platform)
         if specs_for_dep.any?
-          match_current_platform ? specs += specs_for_dep : specs |= specs_for_dep
+          specs += specs_for_dep
```

Comment:
I was looking at this line of code too after chatting with @ngan . `concat` would be faster since it adds to the existing array without creating a new array. 

```suggestion
          specs += specs_for_dep
```

It'd also use less memory, which I only mention because one of the flamegraphs I was looking at did show GC happening during `bundler/setup

Reply:
Yes, I actually tried `concat` too, but it didn't show a significant speed up on my testing, so I left it out for now to keep the patch more focused. But we can add it yeah.

Reply:
Added! It leads to about 10% memory saving on `rails/rails` Gemfile!